src/actions/SettingsActions.tsx

```diff
@@ -298,3 +303,41 @@ export function setSpamFilterOn(spamFilterOn: boolean): ThunkAction<void> {
       .catch(showError)
   }
 }
+
+/**
+ * Toggle the 'Contacts Access' Edge setting. Will request permissions if
+ * toggled on/enabled AND system-level contacts permissions are not granted.
+ * Does NOT modify system-level contacts permissions if toggling the 'Contacts
+ * Access' setting OFF
+ */
+export function setContactsPermissionOn(contactsPermissionOn: boolean): ThunkAction<void> {
+  return (dispatch, getState) => {
```

Comment:
If you make this function `async (dispatch, getState)`, then you could clean up a huge amount of error handling. You would need to add a single `.catch(showError)` on the settings scene to handle the whole mess.